### PR TITLE
Additional Paper App fields required #165475824

### PR DESCRIPTION
**Pull request by LoganArnett**
This PR covers ticket [#165475824](https://www.pivotaltracker.com/story/show/165475824)

Adds some additional required fields and a little custom validation logic. 

Currently waiting for Google Drive access to confirm all fields mentioned in the ticket are included

### CODE REVIEW

**Comment by akegan** [comment]
@LoganArnett are you still waiting for google drive access?

File: app/javascript/components/applications/application_form/DemographicInfoSection.js [hunk 1]
```diff
@@ -10,7 +11,13 @@ let {
   ethnicityOptions
 } = formOptions
 
-const DemographicInfoSection = ({ defaultValues }) => {
+const DemographicInfoSection = ({ values: { demographics } }) => {
+  const validateRelatedField = (validationMessage, relatedField) => {
```
Comment (akegan) [inline app/javascript/components/applications/application_form/DemographicInfoSection.js]: @LoganArnett this actually isn't the desired behavior. It's more like gender_other is required if the leasing agent selects "not listed" for gender. That's an optional piece of this ticket though, don't need to implement it right now.

**Review by akegan — CHANGES_REQUESTED** [review]

File: app/javascript/components/applications/application_form/DemographicInfoSection.js (hunk 1 above)
Comment (LoganArnett) [reply app/javascript/components/applications/application_form/DemographicInfoSection.js]: I missed that that was an option in the `gender` field I can update that if we want for this PR

Comment (akegan) [reply app/javascript/components/applications/application_form/DemographicInfoSection.js]: ok yeah sounds good! then it's just a refactor of what you have already, right?

Comment (LoganArnett) [reply app/javascript/components/applications/application_form/DemographicInfoSection.js]: Added this in

**Review by akegan — APPROVED** [review]

**Comment by akegan** [comment]
awesome. looks good!

**Pull request merged**